distribution/lib/Standard/Test/0.0.0-dev/src/Main.enso

```diff
@@ -551,3 +543,12 @@ run_spec ~behavior =
                 Failure ("An unexpected error was returned: " + err.to_text + '\n' + stack_trace_text)
             _ -> Failure ("An unexpected panic was thrown: " + ex.to_text + '\n' + maybeExc.get_stack_trace_text)
     result
+
+## PRIVATE
+   Reports an unexpected dataflow error has occurred.
+fail_match_on_unexpected_error : Error -> Integer -> Nothing
+fail_match_on_unexpected_error error frames_to_skip =
+    payload = error.catch
+    loc = Meta.get_source_location 1+frames_to_skip
```

Comment:
A random thought in general. WDYT about using `Runtime.get_stack_trace` and walking the stack until we leave this library for the first time? Should be a good proxy for the correct source location.